### ISSUE TITLE
Spy does not show an exception was thrown

### ISSUE THREAD

**Issue by suityou01** [sinonjs/sinon#2262]
**Describe the bug**
I am spying on a function that throws a TypeError. When I assert that the spy threw anything at all it tells me that the spy did not throw any exception. This is patently untrue as evidence by the fact that the catch block handles the exception.

On a side note, I have spent a few hours hunting down code examples of what should be a common use case for this library and I cannot find something that is both "descriptive" and "demonstrative".

A 50$ code example would really really help.

**To Reproduce**
```
it('throws an exception and logs to exceptions file when connection is not available', async () => {
    const mockConnection = {}; //This means the connection object is not available
    const fs = new FormService(mockConnection);
    const form = {};
    let spy = sinon.spy(fs,'addForm'); //This is the function I wish to spy on
    sinon.replace(logger,'error',sinon.fake()); //This works just fine

    try {
        const uid = await fs.addForm(form);
    }
    catch(e)
    {
        console.log(e); //This catch block fires
    }
    sinon.assert.calledOnce(logger.error); //This works just fine
    console.log(spy.threw()); //This returns false! This is a lie
});
```
**Expected behavior**
The spy.threw() method returns true to reflect that actual truth.

**Context (please complete the following information):**

* Library version: ^9.0.2
* Environment: Linux, node 14
* Other libraries you are using: Mocha

**Comment by mantoni** [comment]
Can you post a runnable example of the issue? I tried to reproduce like this, but it prints `true`, as expected:

```js
const sinon = require('sinon');

const object = {
  test: function () {
    throw new TypeError();
  }
};

sinon.spy(object, 'test');

try {
  object.test();
} catch(e) {
}

console.log(object.test.threw());
```

**Comment by mantoni** [comment]
Ah, I see now. You're using `await`, so `fs.addForm` doesn't throw, it returns a promise that gets rejected.

**Comment by suityou01** [comment]
Give me a chance to answer before closing the question would you?

```
function bang(msg){
    throw new TypeError(msg);
}
```

```
let spy = sinon.spy(bang);
    try{
        bang('Yeehaw!');
    } catch(e) {
        console.log(`The exception fired ${e}`);
    }
    console.log(spy.threw());
```

This prints out false

**Comment by suityou01** [comment]
And yes it does throw as my catch block catches it (see OP)...

**Comment by mantoni** [comment]
Yes, now you're calling the original function and not through the spy. Obviously it's throwing, and yes, `spy.threw()` returns `false` because the spy wasn't even called.

**Comment by suityou01** [comment]
Slow down, take a minute and put yourself in my shoes.

Your documentation sucks. 
You asked me to post an example after I already did.
You didn't read my question properly.
You answer me in answers that are too short to mean anything.

PLEASE

Post a detailed example that shows me what the actual F you are talking about.

Thanks

**Comment by suityou01** [comment]
Please also explain how your example goes "through the spy"

```sinon.spy(object, 'test');```

```object.test();```

??

**Comment by mantoni** [comment]
The only thing you're achiving with attacking me is that I regret trying to be helpful. If our documentation sucks, you're welcome to help improving it.

**Comment by suityou01** [comment]
Don't behave in a way that is likely to cause upset and offence and waste the time of others. 

Sent from Yahoo Mail on Android 
 
  On Thu, 28 May 2020 at 15:12, Maximilian Antoni<notifications@github.com> wrote:   


The only thing you're achiving with attacking me is that I regret trying to be helpful. If our documentation sucks, you're welcome to help improving it.

**Comment by mantoni** [comment]
I showed you an example that proves that there is no issue with Sinon. I also explained to you what your actual issue is. I don't see why I shouldn't close this. There is a "reopen" feature that I'm happily utilizing if my assesment was wrong. If you're offended by that, I'm sorry. We can still have a discussion around your problem on a closed issue, but there is no issue with the library, therefore I closed.
